File: frontend/src/types/conference.ts
import { Keynote } from "./keynote";

export interface Conference {
    conferenceId: string;
    title: string;
    type: 'ACADEMIC' | 'COMMERCIAL';
    date: string;
    duration: number;
    registeredCount: number;
    score: number;
    keynoteId: string;
    reviews: Review[];
}

export interface Review {
    reviewId: string;
    date: string;
    text: string;
    stars: number;
    conference: Conference;
}

File: conference-service/src/main/java/com/conference/conference/query/entity/Conference.java
package com.conference.conference.query.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Conference {
    @Id
    private String conferenceId;
    private String title;
    private String type;
    private LocalDateTime date;
    private int duration;
    private int registeredCount;
    private double score;
    private String keynoteId;

    @OneToMany(mappedBy = "conference", cascade = CascadeType.ALL)
    private List<Review> reviews;
}
